conjure-go-client/httpclient/request_params.go

```diff
@@ -102,29 +102,42 @@ func WithRequestBody(input interface{}, encoder codecs.Encoder) RequestParam {
 //	input, _ := os.Open("file.txt")
 //	resp, err := client.Do(..., WithRawRequestBody(input), ...)
 //
-// Deprecated: Retries don't include the body for WithRawRequestBody.
-// Use WithRawRequestBodyProvider for full retry support.
+// Retries don't include the body for WithRawRequestBody.
+// Use WithRawRequestBodyPayload for full retry support.
 func WithRawRequestBody(input io.ReadCloser) RequestParam {
-	return WithRawRequestBodyProvider(func() io.ReadCloser {
-		return input
+	return requestParamFunc(func(b *requestBuilder) error {
+		b.bodyMiddleware.requestInput = RequestBodyStreamOnce(func() io.ReadCloser { return input })
+		b.bodyMiddleware.requestEncoder = nil
+		b.headers.Set("Content-Type", "application/octet-stream")
+		return nil
+	})
+}
+
+// WithBinaryRequestBody sets the request body to the input without encoding.
+// See the documentation and constructors for RequestBody for details.
+func WithBinaryRequestBody(input RequestBody) RequestParam {
```

Comment:
Is there anyway for conjure-go-generated code to call this without introducing a break for consumers?

Reply:
Yes, if we leave all the generated exported API as-is, the generated method impls that currently call `httpclient. WithRequestBody(bodyArg)` could instead call `httpclient.WithBinaryRequestBody(httpclient.RequestBodyStreamWithReplay(bodyArg)`, but we want to let the caller pick their own RequestBody type so I'm recommending we change the type of `bodyArg` from `func() io.ReadCloser` to `httpclient.RequestBody`